controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -151,14 +155,33 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
 }
 
 func (r *StorageClassClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueStorageConsumerRequest := handler.EnqueueRequestsFromMapFunc(
+		func(obj client.Object) []reconcile.Request {
+			annotation := obj.GetAnnotations()
+			if namespace, ok := annotation[StorageClassClaimNamespaceAnnotation]; ok {
+				return []reconcile.Request{{
+					NamespacedName: types.NamespacedName{
+						Name:      obj.GetName(),
+						Namespace: namespace,
+					},
+				}}
+
+			}
+			return []reconcile.Request{}
+		})
```

Comment:
Although we agreed to match the name of the claim with the name of the storage class, this conversion might change in the future and it will be better to avoid writing code based on this conversion. As suggested in the last comment, let's try to capture the namespace name of the claim in the annotation. 

```suggestion
	enqueueStorageConsumerRequest := handler.EnqueueRequestsFromMapFunc(
		func(obj client.Object) []reconcile.Request {
			if annotation, found := annotation[StorageClassClaimAnnotation]; found {
				parts := strings.split(annotation, "/")
				return []reconcile.Request{{
					NamespacedName: types.NamespacedName{
						Name:      parts[0]
						Namespace: parts[1],
					},
				}}
			} else {
				return []reconcile.Request{}
			}
		},
	)
```

Reply:
Done